src/wh_transport_mem.c

```diff
@@ -205,3 +207,37 @@ int wh_TransportMem_RecvResponse(void* c, uint16_t *out_len, void* data)
 
     return 0;
 }
+
+int wh_TransportMem_Cancel(void* c, uint16_t seq)
+{
+    whTransportMemContext* context = c;
+
+    if (    (context == NULL) ||
+            (context->initialized == 0)) {
+        return WH_ERROR_BADARGS;
+    }
+
+    if (*(context->cancel_seq) != 0)
+        return WH_ERROR_NOTREADY;
+
+    /* Check to see if the current response is the different than the request */
+    *(context->cancel_seq) = seq;
+
+    return 0;
+}
+
+int wh_TransportMem_Canceled(void* c, uint16_t* seq)
```

Comment:
same comment as with the callback struct: 

can we rename to something a little more clear and self-describing...perhaps isCanceled or checkCanceledState or something like that? Just something to indicate it is a state query without needing to look at the body of the function and trace the logic through.